Add Python 2.7 support, checkers/linters config, official CC0 1.0 license

DrMattChristian opened the pull request:
A bunch of changes to support Python 2.7 while retaining Python 3 and newer. Ran ct-honeybee through a half dozen Python checkers/linters and resolved errors, warnings and some nitpicks. Only disable certificate verification when connecting to CT logs, not for auditors too. Updated LICENSE file to use the official creative commons zero file so GitHub will auto-detect properly.

I successfully tested my new version on Python 2.7, 3.4 and 3.6.

The diff looks huge but contains a lot of white space, comments/Docstrings, formatting, case and other changes. For a more compact diff you have to use some CLI options:
`git diff -w --minimal --word-diff`

DrMattChristian committed:
Rename COPYING to LICENSE so GitHub Licensee detection will work.

DrMattChristian committed:
Add flake8/pep8 linting tools config file.

DrMattChristian committed:
Add pylint config file, based on template then customized.

DrMattChristian committed:
Add a basic Makefile for checking/linting only. Can be extended for install.

DrMattChristian committed:
Force GitHub to redetect LICENSE file.

Deleted last blank line.

DrMattChristian committed:
Update license to use actual, final CC0 1.0 directly from creative commons GitHub. Removes web site-only not-legal-advice header and minor formatting changes.

DrMattChristian committed:
Merge branch 'master' of https://github.com/DrMattChristian/ct-honeybee

DrMattChristian committed:
Python is required and tested with versions 2.7, 3.4 and 3.6.

DrMattChristian committed:
Major changes for Python 2.7 support, while keeping Python 3+.
Fix dozens of checker/linter errors, warnings and most nitpicks.
Only disable certificate validation when connecting to the CT logs.
Do not monkeypatch ssl, pass in HTTPS context instead.
Simplify and clean up the Base64 regular expression.
Use Docstrings (PEP 0257) instead of only comments.
Capitalize CONSTANTS to resolve linter warnings.
Use this and that instead of a and b to resolve linter warnings.

AGWA commented:
Hi @DrMattChristian, thanks for putting together this thoughtful PR!  I have a few questions to help me better understand your changes:

1. Why is the new CC0 file better than the current one?  The current one is bit-for-bit identical to the one available for download at https://creativecommons.org/2011/04/15/plaintext-versions-of-creative-commons-licenses-and-cc0/ and which is provided expressly for bundling with open source projects.  Additionally, https://wiki.creativecommons.org/wiki/CC0_FAQ#May_I_apply_CC0_to_computer_software.3F_If_so.2C_is_there_a_recommended_implementation.3F recommends naming the file `COPYING` or `COPYING.txt`

2. Why is Python 2 support necessary, when there are only two more years left before it's EOL?  Are there platforms on which Python 3 is still not readily available?

3. Is the new base64 regex correct? It doesn't look like it would match a simple base64 string such as `dGVzdHRlc3QK`.

DrMattChristian commented:
Hi @AGWA,
All good questions, here are my answers.

1. This CC0 file [came directly from creative commons](https://github.com/creativecommons/global-network-strategy/blob/master/LICENSE) and detected by [GitHub Licensee](https://help.github.com/articles/licensing-a-repository/) for searching, display, etc. After I renamed COPYING to LICENSE, I found out that GItHub will [look for COPYING as well](https://github.com/benbalter/licensee/blob/master/lib/licensee/project_files/license_file.rb#L6-L43) so LICENSE can be moved back to COPYING, I did that now.
2. I added Python 2 support for my needs to support CentOS/OL/RHEL 6.x _out of the box_ without additional 3 installs from EPEL/IUS repos. I left full Python 3 compatibility as well so no functionality was "lost". However I understand your point about not looking back, avoiding 2 vs 3 pitfalls and gotchas.
3. Good point, you found a base64 RE matching bug. I reverted to your original base64 RE. One reason why I created an [issue ](https://github.com/DrMattChristian/ct-honeybee/issues/1) in my forked repo for better/further Base64 support. I'm thinking of using a base64 library with more error checking, data validation and automatic tests.

Since ct-honeybee is your own creation, it's up to you which (if any) of my PR changes to accept/merge. If you reject then I can/will maintain my own private, downstream fork for my specific needs.

Thanks!

DrMattChristian committed:
Rename LICENSE file back to COPYING per @AGWA request and CC naming convention.

DrMattChristian committed:
Revert base64 regular expression back to previous/original version.

DrMattChristian committed:
Change line length warning to 110 for reverted base64 RE.
